src/Security/Items.cs

```diff
@@ -762,6 +762,48 @@ public SecRecord (SecKind secKind)
 #endif
 		}
 
+		public SecRecord (SecCertificate certificate) : this (SecKind.Certificate)
+		{
+			SetCertificate (certificate);
+		}
+
+		public SecRecord (SecIdentity identity) : this (SecKind.Identity)
+		{
+			SetIdentity (identity);
+		}
+
+		public SecRecord (SecKey key) : this (SecKind.Key)
+		{
+			SetKey (key);
+		}
+
+		public SecCertificate GetCertificate ()
+		{
+			var kind = queryDict.LowlevelObjectForKey (SecClass.SecClassKey);
+			if (kind != SecClass.Certificate)
+				throw new InvalidOperationException ("This SecRecord is not a Certificate record");
+
+			return GetValueRef <SecCertificate> ();
+		}
+
+		public SecIdentity GetIdentity ()
+		{
+			var kind = queryDict.LowlevelObjectForKey (SecClass.SecClassKey);
+			if (kind != SecClass.Identity)
+				throw new InvalidOperationException ("This SecRecord is not an Identity record");
+
+			return GetValueRef<SecIdentity> ();
+		}
+
+		public SecKey GetKey ()
+		{
+			var kind = queryDict.LowlevelObjectForKey (SecClass.SecClassKey);
+			if (kind != SecClass.Key)
+				throw new InvalidOperationException ("This SecRecord is not a Key record");
```

Comment:
Not critical (and a small optimization) but the check could be refactored into a single `CheckClass (SecClass.*)` method. 

I suspect the reason it was not done might be the exception string ? That could be fixed by making `SecClass` a _smart_ enum (but would require a lot of other code changes). Still having a single string (with a format) would reduce metadata size, so maybe change it to `"SecRecord of incompatible SecClass"` ?